module/VuFind/src/VuFind/Controller/AbstractRecord.php

```diff
@@ -170,7 +170,10 @@ public function addcommentAction()
                 true,
                 $driver
             );
-            $resource->addComment($comment, $user);
+            $commentsService = $this->getDbService(
+                \VuFind\Db\Service\CommentsService::class
+            );
+            $commentsService->addComment($comment, $user, $resource);
```

Comment:
PHPStan will start complaining about a potentially polymorphic call e.g. here at some point, if/when we keep bumping up the level. We'd need to assert that the class returned by `getDbService()` is indeed what we asked for. Wondering if this (adding the new database services) would also be a good opportunity to start adding assertions? Not sure, but just a note.

Reply:
I think this is a good idea; I'll see about adding annotations in my follow-up PR addressing getDbService.